Header

gkalvatcheva opened the pull request:
- header-related styles and variables

Need to finish:
- responsive styles on a separate branch
- possibly move new mixin

gkalvatcheva committed:
- tz flag
- addition of header styles
- addition of one blue color variable
- addition of dimensions variables
- addition of text size variable for the dash title
- increased header height a bit to bring closer to design height. first
trying with slightly lower height than mockup

gkalvatcheva committed:
main nav styling

- added styles for main nav list
- added new dimension variable for nav position
- added new font size variable

gkalvatcheva committed:
- separation of margins and dimensions in variables
- dimension variable value adjustments
- border radius mixin ( considering moving to its own file )

gkalvatcheva committed:
- color for active language selection

uniphil commented on app/scripts/components/boilerplate/header.jsx:
```diff
@@ -10,8 +10,9 @@ const Header = React.createClass({
       <div className="header">
         <div className="logo">
-          <h1><T k="site-name" /></h1>
+          <img src="images/tz-flag.png"/>
```
All `<img>` tags should have an `alt` prop for accessibility. It's actually hard in this case because our `<T>` translation component only renders strings, not props.

@esoliani I made a component for setting translated props, just wondering if you think it's way to ugly and we should do something else. It works like this:

``` js
<div className="logo">
  <TSetChildProp props={{alt: 'site.name'}}>
    <img src="/images/tz-flag.png" />
  </TSetChildProps>
</div>
```

so, that would inject `alt="Water Dashboards"` into the `<img>` tag.

Not sure a better way to do it that would still automatically update the translation when the language is changed.

uniphil replied:
also @gkalvatcheva make paths to static components aboslute, so like `/images/...`.

That way, we can, maybe, later, set up the app routing to use the real URL path (so the app might be loaded from somewhere like `/waterpoints/some-waterpoint-id`), and links like this will still work.

gkalvatcheva replied:
Not adding alt tag for now since we're setting up translation for it. Here is the ticket for the task: https://jira.dgfoundation.org/browse/TSD-227

uniphil commented on app/stylesheets/config/list.scss:
```diff
@@ -15,3 +15,9 @@
   }
 }
+
+// probably should move this to it's own file?
```
yes :)
maybe `box.scss`? or something...

gkalvatcheva replied:
Cool. Added box.css. I like short names. :)

uniphil commented:
looking great!

uniphil commented on app/stylesheets/boilerplate/header.scss:
```diff
@@ -1,10 +1,86 @@
     float: right;
+    margin-top: $nav-top-position;
+
+    & > ul {
```
careful, these styles should go in stylesheets for the child components, not here. Only rules for html in the component should go in its styles.

uniphil commented on app/stylesheets/boilerplate/header.scss:
```diff
@@ -1,10 +1,86 @@
+      }
+    }
+
+  ul.language-selector {
```
move to `language-selector.scss`

gkalvatcheva committed:
- moved border radius mixin to it’s own file
- moved language selector to it’s file

gkalvatcheva committed:
- moved styling specific to the logo section. only left styles related
to position of sections within the header
- added back hover for language selector
- small changes to open data nav
- list adjustments

uniphil commented on app/stylesheets/boilerplate/header.scss:
```diff
@@ -1,10 +1,15 @@
+  padding-left: $header-common-margin;
+  padding-right: $header-common-margin;
+  .logo {
+    float: left;}
```
watch the closing brace position, and spacing between rule blocks

gkalvatcheva replied:
Just updated to remove the lint error.

uniphil commented on app/stylesheets/boilerplate/language-selector.scss:
```diff
@@ -1,10 +1,29 @@
+ul.language-selector {
   @include list-inline();
   @include list-unstyle();
+  @include border-radius(3px);
```
should `3px` be in a variable? we don't have a lot of rounded corners, _maybe_ we can just leave this one hard-coded...

uniphil commented on app/stylesheets/boilerplate/logo.scss:
```diff
@@ -0,0 +1,24 @@
+.header {
+
+  img {
+      float: left;
```
double-indented

gkalvatcheva replied:
cleaned up

uniphil commented on app/stylesheets/boilerplate/logo.scss:
```diff
@@ -0,0 +1,24 @@
+      float: left;
+      font-size: $dash-title;
+      font-weight: $title-weight;
+      line-height: $dash-title + 2;
```
ok for the title to have this `+ 2`, but for other text we can keep `line-height` variables in `type.scss`

gkalvatcheva replied:
@uniphil  Took your advice and added variable. :)

uniphil commented on app/stylesheets/config/box.scss:
```diff
@@ -0,0 +1,4 @@
+@mixin border-radius($radius) {
+  background-clip: padding-box;  /* stops bg color from leaking outside the border: */
```
awesome! :+1: I didn't know about this. Really good to know!

uniphil commented:
cool. couple more small things, plus some lint issues and then this will be good to go.
